extensions/mssql/package.json

```diff
@@ -131,6 +157,66 @@
         }
       }
     },
+    "menus": {
+      "commandPalette": [
+        {
+          "command": "bdc.uploadFiles",
+          "when": "false"
+        },
+        {
+          "command": "bdc.mkdir",
+          "when": "false"
+        },
+        {
+          "command": "bdc.deleteFiles",
+          "when": "false"
+        },
+        {
+          "command": "bdc.previewFile",
+          "when": "false"
+        },
+        {
+          "command": "bdc.saveFile",
+          "when": "false"
+        },
+        {
+          "command": "bdc.copyPath",
+          "when": "false"
+        }
+      ],
+      "objectExplorer/item/context": [
+        {
+          "command": "bdc.uploadFiles",
+          "when": "nodeType=~/^bdc/ && nodeType != bdc:message && nodeType != bdc:file",
+          "group": "1bdc@1"
+        },
+        {
+          "command": "bdc.mkdir",
+          "when": "nodeType=~/^bdc/ && nodeType != bdc:message && nodeType != bdc:file",
+          "group": "1bdc@1"
+        },
+        {
+          "command": "bdc.saveFile",
+          "when": "nodeType == bdc:file",
+          "group": "1bdc@1"
+        },
+        {
+          "command": "bdc.previewFile",
+          "when": "nodeType == bdc:file",
+          "group": "1bdc@2"
+        },
+        {
+          "command": "bdc.copyPath",
+          "when": "nodeType=~/^bdc/ && nodeType != bdc:connection && nodeType != bdc:message",
```

Comment:
Just thinking about this - could we use `mssqlCluster` instead of `bdc` everywhere? If others want to understand the node type it'll help.